src/baf/knot_vector.cc

```diff
@@ -112,6 +112,14 @@ size_t baf::KnotVector::GetNumberOfKnots() const {
   return knots_.size();
 }
 
+int baf::KnotVector::GetNumberOfDifferentKnots() const {
+  int number = 1;
```

Comment:
Currently, you add 1 here, as you do not initialize with 0. I would consider to replace `int number = 1` by `int number_non_zero_knot_spans = 0` and add 1 when returning.